src/components/secrets/Secret.tsx

```diff
@@ -484,16 +477,17 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
             let isValid = !isESO ? secretDataArray.reduce((isValid, s) => {
                 isValid = isValid && !!s.fileName && !!s.name
                 return isValid
-            }, true) :
+            }, !!secretData.length) :
             secretDataArray?.reduce((isValid, s) => {
                 isValid = isValid && !!s.secretKey && !!s.key && 
                 (hasProperty(externalType) ? !!s.property : true)
                 return isValid
-            }, true)
+            }, !!secretStore && !!esoSecretData?.length)
+
             if (!isValid) {
                 !isESO
                     ? toast.warn('Please check key and name')
```

Comment:
Please use if else statement here for more readiability

Reply:
done